### ISSUE TITLE
How to run a single tag only?

### ISSUE THREAD

**Issue by tgsmith61591** [cucumber-rs/cucumber#156]
The [doc suggests](https://cucumber-rs.github.io/cucumber/current/Features.html#cli-options) running a subset of tags works. However, I cannot get this to work in my Cucumber project:

```bash
❯ cargo test -- --tags='@music'
...
error: Unrecognized option: 'tags'
```

How can I select which tag to run?

**Comment by ilslv** [comment]
@tgsmith61591 first of all please provide version you are using

Our docs say following:

> Example with tag expressions for filtering Scenarios:

```
cargo test --test <test-name> -- --tags='@cat or @dog or @ferris'
```

Note existence of `--test <test-name>`. It's required to specify single Rust test to run, while running `cargo test -- --tags='@cat or @dog or @ferris'` tries to execute all existing tests, which can be unit or doc tests. They don't know nothing about cucumber cli options.

**Comment by tgsmith61591** [comment]
I'm on version 0.9.0. Once I provided `<test-name>`, I've begun to get a different error.

Here is my `Cargo.toml`:

```toml
[package]
name = "foo_package"
# ...

[[test]]
name = "cucumber"
harness = false

# ...

[dev-dependencies]
cucumber_rust = "0.9.0"
tokio = { version = "1.2.0", features = ["rt-multi-thread", "macros", "time"] }

```

To trigger the test: `$ cargo test --test cucumber -- --tags='@music'`, results in a new error:

```bash
error: Found argument '--tags' which wasn't expected, or isn't valid in this context
```

**Comment by ilslv** [comment]
@tgsmith61591 documentation you referred to is for `v0.10` and later. Since `v0.9` crate even got renamed from `cucumber_rust` to `cucumber` and many more changes that you can see in changelog. 

You have 2 options: 
1. Upgrade to the new version and use redesigned cli options
2. Use `v0.9` cli options. 
  There was no documentation for it, but I remember it being present. #67 suggests the same. From what I can remember cli options where enabled by calling `.cli()` on `Cucumber`. Then you could filter scenarios by name, but not using tag. I don't quite remember cli option, please use `cargo test --test cucumber -- --help` to see everything that is enabled.

**Comment by tgsmith61591** [comment]
Thanks @ilslv! By upgrading to 0.10.2 I was able to get the `--tags` argument to work. However, I should point out it seems parsing tags at the `Feature` level does not work:

```gherkin
# does not work
@my_foo_tag
Feature: Test all foo

  Scenario: ...

  Scenario: ...

```

Gives:

```
[Summary]
0 features
0 scenarios
0 steps
```

vs.

```gherkin
# does work
Feature: test all foo

  @my_foo_tag
  Scenario: ...

  @my_foo_tag
  Scenario: ...

```

Gives:

```
[Summary]
1 feature
48 scenarios (48 passed)
249 steps (249 passed)
```

It's not too painful to add the tags directly to scenarios, but may be worth looking into...

**Comment by ilslv** [comment]
@tgsmith61591 yes, `Feature` tags are ignored by design. Every other implementation I saw does the same thing. This is mainly done to avoid situations when in big `.feature` file you are adding a new `Scenario`, but its tags get overwritten by `Feature` ones. 

To be fair, you can use `Feature` and `Rule` tags inside [`Cucumber::filter_run(_and_exit)?`][1] methods, where you have more control comparing to just `CLI` options. 



[1]: https://docs.rs/cucumber/0.10.2/cucumber/struct.Cucumber.html#method.filter_run_and_exit

**Comment by tgsmith61591** [comment]
Understood. For what it's worth, [Godog](https://github.com/cucumber/godog) (Golang) and [behave](https://github.com/behave/behave) (Python) both support this behavior. It's not a big deal, but makes portability a bit of an issue.

In any case, thanks for your help!

**Comment by tyranron** [comment]
@tgsmith61591 

> Understood. For what it's worth, [Godog](https://github.com/cucumber/godog) (Golang) and [behave](https://github.com/behave/behave) (Python) both support this behavior. It's not a big deal, but makes portability a bit of an issue.

Now supported, and will be released in `0.11`.
